internal/service/project/model_project.go

```diff
@@ -202,3 +206,32 @@ func UpdateProjectBool(plan, state types.Bool, setting **bool) bool {
 	}
 	return false
 }
+
+func NewTFTags(tags *[]admin.ResourceTag) types.Map {
+	if tags == nil {
+		return types.MapNull(types.StringType)
+	}
+
+	typesTags := make(map[string]attr.Value, len(*tags))
+	for _, tag := range *tags {
+		typesTags[tag.Key] = types.StringValue(tag.Value)
+	}
+	return types.MapValueMust(types.StringType, typesTags)
+}
+
+func NewAdminTags(ctx context.Context, tags types.Map) *[]admin.ResourceTag {
+	if tags.IsNull() || len(tags.Elements()) == 0 {
+		return &[]admin.ResourceTag{}
+	}
+	elements := make(map[string]types.String, len(tags.Elements()))
+	// todo: want to improve this.
```

Comment:
I have an idea of improving this. But I think it belongs in a separate PR

Reply:
avoid this generic comment in the code. Try to be more specific and preferably link to a CLOUDP jira issue

Reply:
also I am curious of what you'd like to improve and why we can't just do it now :)

Reply:
I'll present it in the next tech meeting (`API-> TF` & `TF -> API` model conversions). 
For now,  I removed the comment.